### DISCUSSION TITLE
dial tcp lookup image-registry.openshift-image-registry.svc on 10.10.8.132-53 no such host

### DISCUSSION THREAD

**Discussion by bobby0724** [okd-project/okd#694] [Q&A]
I have successfully completed installation on bare metal using UPI and I have setup NFS share for the internal registry

I try to deploy a sample application and I am getting the following error

seems that the OKD is querying  my DNS instead of the internal DNS service is there any way i an fix this ?

Failed to pull image "image-registry.openshift-image-registry.svc:5000/basictest/httpd-sample:latest": rpc error: code = Unknown desc = error pinging docker registry image-registry.openshift-image-registry.svc:5000: Get "https://image-registry.openshift-image-registry.svc:5000/v2/": dial tcp: lookup image-registry.openshift-image-registry.svc on 10.10.8.132:53: no such host

**Comment by bobby0724** [comment]
@fortinj66 is there a chance you can help me to check this issue

**Reply by gudroot** [reply]
To me this looks like another case of the search=dot problem from issue https://github.com/openshift/okd/issues/648#issuecomment-853111092

**Comment by fortinj66** [comment]
Is your internal registry running?  I was able to recreate this on a new UPI/DHCP cluster but the issue really was that the internal registry was not configured/Running.  Once that was completed internal DNS was fine...

check to see if you have a service entry for the registry

```
# oc get service -n openshift-image-registry
NAME                      TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)     AGE
image-registry            ClusterIP   172.30.181.130   <none>        5000/TCP    25m
image-registry-operator   ClusterIP   None             <none>        60000/TCP   103m
```

**Reply by bobby0724** [reply]
> Is your internal registry running? I was able to recreate this on a new UPI/DHCP cluster but the issue really was that the internal registry was not configured/Running. Once that was completed internal DNS was fine...
> 
> check to see if you have a service entry for the registry
> 
> ```
> # oc get service -n openshift-image-registry
> NAME                      TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)     AGE
> image-registry            ClusterIP   172.30.181.130   <none>        5000/TCP    25m
> image-registry-operator   ClusterIP   None             <none>        60000/TCP   103m
> ```

Yes it is running actually sending the hostname and dns from DHCP fix the issue, but as you said FCOS should be able to resolve the name

at some point i end up with all the nodes with localhost as hotname and nothing works, after setting hostname in the DHCP reservations everything go back to normal

**Reply by fortinj66** [reply]
Yep, don't depend on reverse DNS host assignment...

**Comment by fortinj66** [comment]
OK, I can reproduce with static IP addresses configured on a vanilla FCOS 34 install

```
[core@worker0 ~]$ cat /etc/resolv.conf | grep -v "\#"

nameserver 10.99.111.1
nameserver 10.99.111.2
search .
```

**Comment by fortinj66** [comment]
Potential workaround/fix for 'search .' issue
 
https://github.com/openshift/okd-machine-os/pull/156
